3-UsingAPIs/Week2/assignment/ex2-pokemonApp/index.js

```diff
@@ -21,18 +21,84 @@ Use async/await and try/catch to handle promises.
 Try and avoid using global variables. As much as possible, try and use function 
 parameters and return values to pass data back and forth.
 ------------------------------------------------------------------------------*/
-function fetchData(/* TODO parameter(s) go here */) {
-  // TODO complete this function
-}
+const body = document.querySelector('body');
+
+
+async function fetchData(url) {
+  try {
+    const response = await fetch(url)
+    if (!response.ok) {
+      throw new Error("SOME ERROR WITH API" +  response.status);
+    }    
+    return response.json();
+
+   } catch (error) {
+    console.log(error);
+    throw error;
+   }
+  }
+
+function fetchAndPopulatePokemons(data) {
+  //createElements
+  const select = document.createElement("select");
+  const img = document.createElement('img');
+  
+
+  // appendChild
+  body.appendChild(select);
+  body.appendChild(img);
+
+ // add options
+  const {results} = data;
+  results.map((pokemon) => {
+    const namePokemon = pokemon.name;
+    const option = document.createElement("option");
+    option.innerHTML = namePokemon;
+    select.appendChild(option);
+  })
 
-function fetchAndPopulatePokemons(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+   console.log(data); 
 }
 
-function fetchImage(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+
+async function fetchImage(data) {
+  const select = document.querySelector('select');
+  const {results} = data;
+  const currentPokemon = results.find(pokemon => select.value === pokemon.name);
+  const url = currentPokemon.url;
```

Comment:
If you had stored the pokemon url in `option.value` as I suggested above, you could simplify the code here:
 
```suggestion
  const url = select.value;
```